packages/react-components/react-storybook/src/decorators/withFluentVrTestVariants.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { FluentProvider } from '@fluentui/react-provider';
+import { teamsHighContrastTheme, webDarkTheme, webLightTheme } from '@fluentui/react-theme';
+import { makeDecorator } from '@storybook/addons';
+import type { StoryContext } from '@storybook/addons';
+
+export const DARK_MODE = 'DarkMode';
+export const HIGH_CONTRAST = 'HighContrast';
+export const RTL = 'Rtl';
+
+// MakeDecoratorOptions is not an exported interface by storybook so had to include necessary options
+interface AssertedMakeDecoratorOptions {
+  name: string;
+  parameterName: string;
+  skipIfNoParametersOrOptions?: boolean;
+  wrapper: (
+    storyFn: (context: StoryContext) => React.ReactNode,
+    context: StoryContext,
+    settings: {
+      parameters: typeof DARK_MODE | typeof HIGH_CONTRAST | typeof RTL;
+    },
+  ) => React.ReactNode;
+}
+
+type AssertedMakeDecorator = ({
+  name,
+  parameterName,
+  skipIfNoParametersOrOptions,
+  wrapper,
+}: AssertedMakeDecoratorOptions) => () => React.ReactNode;
+
+export const withFluentVrTestVariants = ((makeDecorator as unknown) as AssertedMakeDecorator)({
+  name: 'withFluentVrTestVariants.',
+  parameterName: 'vrTestVariant',
+  skipIfNoParametersOrOptions: true,
+  wrapper: (storyFn, context, { parameters }) => {
+    if (parameters === RTL) {
+      return (
+        <FluentProvider dir={'rtl'} theme={webLightTheme}>
+          {storyFn(context)}
+        </FluentProvider>
+      );
+    }
+
+    if (parameters === DARK_MODE) {
+      return <FluentProvider theme={webDarkTheme}>{storyFn(context)}</FluentProvider>;
+    }
+
+    if (parameters === HIGH_CONTRAST) {
+      return <FluentProvider theme={teamsHighContrastTheme}>{storyFn(context)}</FluentProvider>;
+    }
+
+    return storyFn(context);
```

Comment:
why return without a provider wrapper at all ? is there already another provider that's wrapping everything ? if so let's do a bit of refactoring so that we only add the provider in one place

Reply:
^ great call, lets do it ! #DRY

Reply:
Yeah I considered doing this but as it stands, the withFluentProvider decorator doesn’t just wrap a story function in FluentProvider, but it also wraps it in a container with custom styling. I wanted to avoid a bunch of Screener change churn that this would cause which is why I opted for a separate decorator.

Reply:
in that case we could remove the custom styling into its own decorator ?

Reply:
> I wanted to avoid a bunch of Screener change churn

wondering what churn are you referring to ?

Reply:
Since an extra div with styling is essentially added to each story by the `withFluentProvider ` decorator, when each story is converted to CSF , Screener will detect each one as "Changed". I'm trying to avoid a bunch of unnecessary Screener changes when I refactor to CSF to simplify the review process and to not introduce random screener issues

Reply:
> in that case we could remove the custom styling into its own decorator ?

With the way it's implemented right now, that would just seem redundant? Even if moved as a separate decorator, it would still need a `FluentProvider` regardless. 
```
export const withFluentProvider = (StoryFn: () => JSX.Element, context: FluentStoryContext) => {
  const { theme } = getActiveFluentTheme(context.globals);

  return (
    <FluentProvider theme={theme}>
      <FluentExampleContainer theme={theme}>{StoryFn()}</FluentExampleContainer>
    </FluentProvider>
  );
};
```

Reply:
I wonder if we could utilize the storybook context or globals a bit more, `withFluentProvider` could maybe have like `mode` configuration where the VR tests would configure a mode that would not inject padding and theme background ?